src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SupplierAssessmentService.kt

```diff
@@ -27,4 +32,29 @@ class SupplierAssessmentService(
     )
     return referralRepository.save(referral)
   }
+
+  fun updateSupplierAssessmentAppointment(
+    referral: Referral,
+    durationInMinutes: Int,
+    appointmentTime: OffsetDateTime,
+    createdByUser: AuthUser
+  ): SupplierAssessment {
+    if (referral.supplierAssessment!!.appointments.size == 0) {
```

Comment:
would be good to have more explicit error handling around referral.supplierAssessment

either delegate this to the caller by taking a not null supplierAssessment in the signature, or throw an error with an explicit message like "no supplier assessment for rererral id ${referral.id}"